common/src/main/java/com/lightstep/tracer/shared/AbstractTracer.java

```diff
@@ -293,29 +296,29 @@ public void disable() {
     return this.new SpanBuilder(operationName);
   }
 
-  public void inject(io.opentracing.SpanContext spanContext, Object carrier) {
+  public <C> void inject(io.opentracing.SpanContext spanContext, Format<C> format, C carrier) {
     SpanContext lightstepSpanContext = (SpanContext)spanContext;
-    if (Propagator.TEXT_MAP.matchesInjectCarrier(carrier)) {
-      Propagator.TEXT_MAP.inject(lightstepSpanContext, carrier);
-    } else if (Propagator.HTTP_HEADER.matchesInjectCarrier(carrier)) {
-      Propagator.HTTP_HEADER.inject(lightstepSpanContext, carrier);
-    } else if (Propagator.BINARY.matchesInjectCarrier(carrier)) {
+    if (format == Format.Builtin.TEXT_MAP) {
+      Propagator.TEXT_MAP.inject(lightstepSpanContext, (TextMap)carrier);
```

Comment:
these casts are a cheesy way of doing this, but building a full-on registry for two types seems ~ridiculous...